zerver/lib/streams.py

```diff
@@ -337,6 +337,16 @@ def access_stream_by_name(user_profile: UserProfile,
                                             allow_realm_admin=allow_realm_admin)
     return (stream, recipient, sub)
 
+def access_web_public_stream(stream_id: int, realm: Realm) -> Optional[Stream]:
+    try:
+        stream = get_stream_by_id_in_realm(stream_id, realm)
+    except Stream.DoesNotExist:
+        raise JsonableError(_("Invalid stream id"))
+
+    if not stream.is_web_public:
+        return None
+    return stream
```

Comment:
This should just raise `JsonableError` for the other case and return `Stream`; there's no reason we should make the caller responsible for also checking for `None`.